src/Controller/ArticlesController.php

```diff
@@ -377,25 +378,47 @@ private function createFirstPage(Request $request, string $id, array $arguments)
                 }
 
                 if ($item->getReviewers()->notEmpty()) {
-                    $infoSections[] = ArticleSection::basic(
-                        'Reviewing Editor',
-                        3,
-                        $this->render(
-                            Listing::ordered(
-                                $item->getReviewers()
-                                    ->map(function (Reviewer $reviewer) {
-                                        $parts = [$reviewer->getPreferredName(), $reviewer->getRole()];
-
-                                        foreach ($reviewer->getAffiliations() as $affiliation) {
-                                            $parts[] = $affiliation->toString();
-                                        }
-
-                                        return implode(', ', $parts);
-                                    })
-                                    ->toArray()
-                            )
-                        )
-                    );
+                    $roles = $item->getReviewers()
+                        ->reduce(function (array $roles, Reviewer $reviewer) {
+                            $entry = $reviewer->getPreferredName();
+
+                            foreach ($reviewer->getAffiliations() as $affiliation) {
+                                $entry .= ", {$affiliation->toString()}";
+                            }
+
+                            $roles[$reviewer->getRole()][] = $entry;
+
+                            return $roles;
+                        }, []);
+
+                    uksort($roles, function (string $a, string $b) : int {
+                        if (false !== stripos($a, 'Senior')) {
+                            return -1;
+                        }
+                        if (false !== stripos($b, 'Senior')) {
+                            return 1;
+                        }
+                        if (false !== stripos($a, 'Editor')) {
+                            return -1;
+                        }
+                        if (false !== stripos($b, 'Editor')) {
+                            return 1;
+                        }
+
+                        return 0;
+                    });
```

Comment:
Probably a better way to do this.

Reply:
Map the possible roles to numbers, compare those? It seems the order being requested is transitive, so it should work as long as there are not too many roles.